### PR TITLE
Integrating Google Vision's text recognition in Android

### PR DESCRIPTION
**Pull request by alculquicondor**
I need OCR for a project I'm currently working on. I decided to add the changes to the library itself, since I didn't find a way to add custom detection mechanisms and I also would like to contribute.

It is already working, but I would like some feedback, would that be code structure or other missing things (a js type for TextBlock maybe?). Once it is settled, I can document too.

PS: I saw the reverted merge regarding OpenCV, which was also implementing text detection. Is this the direction the project is going? (CC @ma-pe).

### CODE REVIEW

**Comment by ma-pe** [comment]
No, OpenCV is not the direction the project is going.

I accidentally merged my fork, that implements text block detection, but not text recognition.

**Comment by sibelius** [comment]
we would like to support both, but in an optional way

we need to investigate this more https://github.com/react-native-community/react-native-camera/issues/1262

**Comment by sibelius** [comment]
@alculquicondor could u take a stab on iOS code?

**Comment by alculquicondor** [comment]
Sadly no. I don't program for iOS and Google Vision's text recognition is Android only.

**Comment by sibelius** [comment]
could u add this to the docs as well?

**Comment by alculquicondor** [comment]
Done. Wasn't there a RNCameraExample project?

File: src/RNCamera.js [hunk 1]
```diff
@@ -58,6 +58,7 @@ type PropsType = ViewPropTypes & {
   autoFocus?: string | boolean | number,
   faceDetectionClassifications?: number,
   onFacesDetected?: ({ faces: Array<TrackedFaceFeature> }) => void,
+  onTextRecognized?: Function,
```
Comment (sibelius) [inline src/RNCamera.js]: can u improve this flowtype?

what is the shape of this function?

Comment (alculquicondor) [reply src/RNCamera.js]: Done

**Comment by jgfidelis** [comment]
Looks nice. Could you take a screenshot of it working?

For iOS there is a built-in solution, looks like... https://www.appcoda.com/vision-framework-introduction/

**Pull request merged**